docs/sources/reference/components/otelcol/otelcol.processor.interval.md

```diff
@@ -0,0 +1,169 @@
+debug_metrics | [debug_metrics][] | Configures the metrics that this component generates to monitor its state. | no
+
+[output]: #output-block
+[debug_metrics]: #debug_metrics-block
+
+### output block
+
+{{< docs/shared lookup="reference/components/output-block.md" source="alloy" version="<ALLOY_VERSION>" >}}
+
+### debug_metrics block
+
+{{< docs/shared lookup="reference/components/otelcol-debug-metrics-block.md" source="alloy" version="<ALLOY_VERSION>" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name    | Type               | Description
+--------|--------------------|-----------------------------------------------------------------
+`input` | `otelcol.Consumer` | A value that other components can use to send telemetry data to.
+
+`input` accepts `otelcol.Consumer` data for metrics.
+
+## Component health
+
+`otelcol.processor.interval` is only reported as unhealthy if given an invalid configuration.
+
+## Debug information
+
+`otelcol.processor.interval` does not expose any component-specific debug information.
+
+## Example
+
+This example receives OTLP metrics and aggregates them for 30s before sending to the next exporter.
+
+```alloy
+otelcol.receiver.otlp "default" {
+  grpc { ... }
+  http { ... }
+
+  output {
+    metrics = [otelcol.processor.interval.default.input]
+  }
+}
+
+otelcol.processor.interval "default" {
+  interval = "30s"
+  output {
+    metrics = [otelcol.exporter.otlphttp.grafana_cloud.input]
+  }
+}
+
+otelcol.exporter.otlphttp "grafana_cloud" {
+  client {
+    endpoint = "https://otlp-gateway-prod-gb-south-0.grafana.net/otlp"
+    auth     = otelcol.auth.basic.grafana_cloud.handler
+  }
+}
+
+otelcol.auth.basic "grafana_cloud" {
+  username = env("GRAFANA_CLOUD_USERNAME")
+  password = env("GRAFANA_CLOUD_API_KEY")
+}
+```
+
+
+| Timestamp | Metric Name  | Aggregation Temporarility | Attributes        | Value |
+| --------- | ------------ | ------------------------- | ----------------- | ----: |
+| 0         | test_metric  | Cumulative                | labelA: foo       |   4.0 |
+| 2         | test_metric  | Cumulative                | labelA: bar       |   3.1 |
+| 4         | other_metric | Delta                     | fruitType: orange |  77.4 |
+| 6         | test_metric  | Cumulative                | labelA: foo       |   8.2 |
+| 8         | test_metric  | Cumulative                | labelA: foo       |  12.8 |
+| 10        | test_metric  | Cumulative                | labelA: bar       |   6.4 |
+
+The processor would immediately pass the following metrics to the next processor in the chain
+
+| Timestamp | Metric Name  | Aggregation Temporarility | Attributes        | Value |
+| --------- | ------------ | ------------------------- | ----------------- | ----: |
+| 4         | other_metric | Delta                     | fruitType: orange |  77.4 |
+
+Because it's a Delta metric.
```

Comment:
```suggestion
The processor immediately passes the following metric to the next processor in the chain because it is a Delta metric.

| Timestamp | Metric Name  | Aggregation Temporarility | Attributes        | Value |
| --------- | ------------ | ------------------------- | ----------------- | ----: |
| 4         | other_metric | Delta                     | fruitType: orange |  77.4 |

```